NuRadioReco/modules/channelGalacticNoiseAdder.py

```diff
@@ -0,0 +1,226 @@
+                ax_2.grid()
+                ax_2.set_yscale('log')
+                ax_3.grid()
+                ax_3.set_yscale('log')
+                ax_4.grid()
+                ax_4.set_yscale('log')
+                ax_1.set_xlabel('f [MHz]')
+                ax_2.set_xlabel('f [MHz]')
+                ax_3.set_xlabel('f [MHz]')
+                ax_4.set_xlabel('f [MHz]')
+                ax_1.set_ylabel('T [K]')
+                ax_2.set_ylabel('S [W/m²/MHz]')
+                ax_3.set_ylabel('E [V/m]')
+                ax_4.set_ylabel('U [V]')
+                ax_4.plot(channel.get_frequencies() / units.MHz, np.abs(channel.get_frequency_spectrum()), c='C0')
+                ax_4.set_ylim([1.e-8, None])
+                fig1 = plt.figure()
+                ax1 = fig1.add_subplot(211)
+                ax2 = fig1.add_subplot(212)
+                ax1.plot(channel.get_times(), channel.get_trace(), label='original trace')
+                ax2.plot(channel.get_frequencies() / units.MHz, np.abs(channel.get_frequency_spectrum()), label='original spectrum')
+                ax1.grid()
+                ax2.grid()
+            for i_zenith, zenith in enumerate(self.__zenith_sample):
+                solid_angle = (np.cos(zenith) - np.cos(zenith + self.__delta_zenith)) * self.__delta_azimuth
+                for i_azimuth, azimuth in enumerate(self.__azimuth_sample):
+                    temperature_interpolator = scipy.interpolate.interp1d(self.__interpolaiton_frequencies, np.log10(noise_temperatures[:, i_zenith, i_azimuth]), kind='quadratic')
+                    noise_temperature = np.power(10, temperature_interpolator(freqs[passband_filter]))
+                    S = (2. * scipy.constants.Boltzmann * (freqs[passband_filter] / units.Hz)**2 / scipy.constants.c**2 * (noise_temperature / units.kelvin) * solid_angle) * (units.watt / units.m**2 / units.Hz)
+                    S[np.isnan(S)] = 0
+                    S_per_bin = S * d_f
+                    flux_sum += S_per_bin
+                    E = np.sqrt(S_per_bin / (units.watt / units.m**2) / (scipy.constants.c * scipy.constants.epsilon_0)) / (d_f) * units.V / units.m
+                    if self.__debug:
+                        ax_1.scatter(self.__interpolaiton_frequencies / units.MHz, noise_temperatures[:, i_zenith, i_azimuth] / units.kelvin, c='k', alpha=.01)
+                        ax_1.plot(freqs[passband_filter] / units.MHz, noise_temperature, c='k', alpha=.02)
+                        ax_2.plot(freqs[passband_filter] / units.MHz, S_per_bin / d_f / (units.watt / units.m**2 / units.MHz), c='k', alpha=.02)
+                        ax_3.plot(freqs[passband_filter] / units.MHz, E / (units.V / units.m), c='k', alpha=.02)
+
+                    noise_spectrum = np.zeros((3, freqs.shape[0]), dtype=np.complex)
+                    phases = np.random.uniform(0, 2. * np.pi, len(S))
+                    polarizations = np.random.uniform(0, 2. * np.pi, len(S))
```

Comment:
Yes, it's probably better to assign the polarizations in the time domain. But I doubt the effect is going to be noticeable.

Reply:
That doesn't seem right. Wouldn't that result in a polarized wave with a polarization angle of 45°? Basically we set E_theta=E_phi

Reply:
it is not exactly that, but I think the power is distributed equally to eTheta and ePhi. 

What is implemented right now would give a constant polarization over the full recored. I don't think this is the case. The radiation comes from many many sources with random polarization. If you sum everything up, both polarization components have equal power.

Reply:
Maybe @anelles remembers how it was done in the Lofar analysis? Maybe instead the phi and theta component get different random phases?

Reply:
I don't think I understand what you mean. You noticed that the polarization is a different random number for every bin, right?

Reply:
oh I missed that. But doesn't this introduce extremely high frequencies? From one bin to the next you can jump from +1 to -1 and back. Isn't this like generating noise in the time domain which we found lead to wrong results because it does not observe the bandwidth limitations?

Reply:
Then assigning polarization in the frequency domain was probably the right way to do it

Reply:
Okay, I changed this back to the frequency domain.